webapp/src/desktop_notifications.ts

```diff
@@ -0,0 +1,26 @@
+import {Channel} from '@mattermost/types/channels';
+import {Post} from '@mattermost/types/posts';
+
+import {DesktopNotificationArgs, Store} from 'src/types/mattermost-webapp';
+
+import {ringingEnabled} from 'src/selectors';
+import {isDmGmChannel} from 'src/utils';
+
+export function desktopNotificationHandler(store: Store, post: Post, channel: Channel, args: DesktopNotificationArgs): {error?: string, args?: DesktopNotificationArgs} {
+    if (args.notify) {
+        // Calls will notify if:
+        //  1. it's a custom_calls post (call has started)
+        //  2. in a DM or GM channel
+        //  3. calls is enabled and is v0.17.0+ (it is if this is running)
+        //  4. calls ringing is enabled on the server
+
+        // @ts-ignore our imported webapp types are old
```

Comment:
What would it take to update them?

Reply:
We'd have to rearrange webapp and export types correctly in an npm package. I've talked to Harrison about it, probably 6 months ago, and he said it was something he definitely wants to do. But given recent org changes, I'm not sure if it's on the roadmap. I could take that on, if we can fit it in between a feature?

Reply:
Thanks. Not urgent, I was mostly looking to understand what kind of effort was needed and potentially get a ticket open.

Reply:
A ticket to estimate the effort: https://mattermost.atlassian.net/browse/MM-53667